src/api/app/models/bs_request_action.rb

```diff
@@ -920,6 +920,8 @@ def expand_targets(ignore_build_state)
   end
 
   def source_access_check!
+    return if Project.is_remote_project?(source_project, true)
```

Comment:
hm, did you benchmark this? IIRC it can be relative expensive, so we did this only when needed in the past